frontend/packages/console-app/src/actions/creators/hpa-factory.ts

```diff
@@ -93,22 +88,26 @@ type DeploymentActionExtraResources = {
 };
 
 export const useHPAActions = (kindObj: K8sKind, resource: K8sResourceKind) => {
-  const [namespace] = useActiveNamespace();
+  const namespace = resource?.metadata?.namespace;
+
   const watchedResources = React.useMemo(
-    () => ({
-      hpas: {
-        isList: true,
-        kind: HorizontalPodAutoscalerModel.kind,
-        namespace,
-        optional: true,
-      },
-      csvs: {
-        isList: true,
-        kind: referenceForModel(ClusterServiceVersionModel),
-        namespace,
-        optional: true,
-      },
-    }),
+    () =>
+      namespace
+        ? {
+            hpas: {
+              isList: true,
+              kind: HorizontalPodAutoscalerModel.kind,
+              namespace,
+              optional: true,
+            },
+            csvs: {
+              isList: true,
+              kind: referenceForModel(ClusterServiceVersionModel),
+              namespace,
+              optional: true,
+            },
+          }
+        : null,
```

Comment:
At a second glance, it looks like `useK8sWatchResources` might not handle a null arg. You might want to test and make sure this works. You might have to pass in an empty object instead of null;

Reply:
> At a second glance, it looks like `useK8sWatchResources` might not handle a null arg. You might want to test and make sure this works. You might have to pass in an empty object instead of null;

@jon Wondering if we need to worry about handling undefined `namespace` in this case. The HPA and PDB are created in namespaces. Also, I have used `useK8sWatchResources` in a couple places, and looking thro our code base I haven't found a single instance of such. Help me to understand why these cases are different.

Reply:
Hmm, yeah, I see what you're talking about. There are a few places that are similar in logic to this hook. I wasn't really aware of the other similar use cases and I don't know that this case is necessarily different. I'm curious how `useK8sWatchResources` and `useK8sWatchResource` behave when you try to watch a list of namespaced resources and don't provide a namespace. That being said, if this is working as expected in the many other places it's used, I won't hold this PR up for it. It's probably okay to just remove the conditional here.